### PR TITLE
Tokenizer Model

### PR DESCRIPTION
**Pull request by Weiqi97**
### Converted Tokenizer into the MVC structure
- Added `script_tokenizer.js` to EsLint check.
- Removed deprecated DTM functions.
- Removed option to allow users view files as rows. (They can still download in that orientation.)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,6 +37,7 @@ matrix:
         - ./node_modules/.bin/eslint ./js/scripts_similarity.js
         - ./node_modules/.bin/eslint ./js/scripts_dendrogram.js
         - ./node_modules/.bin/eslint ./js/scripts_rwanalysis.js
+        - ./node_modules/.bin/eslint ./js/scripts_tokenizer.js
```
Comment (chantisnake) [inline .travis.yml]: Why not use a regex to match all the scripts?

File: lexos/models/tokenizer_model.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    """A typed tuple that holds all the tokenizer test options."""
+
+    token_type_str: str
+    doc_term_matrix: pd.DataFrame
+    front_end_option: TokenizerOption
+    id_temp_label_map: IdTempLabelMap
+
+
+class TokenizerModel(BaseModel):
+    """This is the class to generate statistics of the input file.
```
Comment (chantisnake) [inline lexos/models/tokenizer_model.py]: I think you forget to change this

File: lexos/models/tokenizer_model.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    def _token_type_str(self) -> str:
+        """:return: A string that represents the token type used."""
+        if self._test_token_type_str is not None:
+            return self._test_token_type_str
+        else:
+            # Get dtm front end options.
+            dtm_options = MatrixReceiver().options_from_front_end()
+            # Get the correct current type.
+            token_type = dtm_options.token_option.token_type
+            return "Terms" if token_type == "word" else "Characters"
```
Comment (chantisnake) [inline lexos/models/tokenizer_model.py]: What are the acceptable values that token type can hold?

File: .travis.yml (hunk 1 above)
Comment (Weiqi97) [reply .travis.yml]: I think we haven't made all the files to pass EsLint yet...

Comment (chantisnake) [reply .travis.yml]: Ohohoh, yeah, sure.

Comment (chantisnake) [reply .travis.yml]: Nope, no need to sorry, things need to be done one at a time. I understand.

**Comment by chantisnake** [comment]
Looks okay to my pity human eyes.

**Pull request merged**